### ISSUE TITLE
getFullMessage() does not display all errors on "key" checks

### ISSUE THREAD

**Issue by gabrielc** [Respect/Validation#179]
Please consider the following code:

```php
$config = array(
    'host'      => 'my_host',
    'user'      => 'my_user',
    'password'  => 'my_password',
    'schema'    => 'my_schema',
);

$validator = v::arr()
    ->setName('Settings')
    ->key('host', v::string())
    ->key('user', v::string())
    ->key('password', v::string())
    ->key('schema', v::string());

try {
    $validator->assert($config);
} catch(\InvalidArgumentException $e) {
    echo $e->getFullMessage().PHP_EOL;
}
```

In this setup everything runs fine, the output is empty;

If I remove only one line in the `$config` array:
```php
$config = array(
    'host'      => 'my_host',
    'password'  => 'my_password',
    'schema'    => 'my_schema',
);
```

The output is:
```
\-These rules must pass for Settings
```

It does not mention the missing `user` key.

If I remove two keys in the `$config` array:
```php
$config = array(
    'host'      => 'my_host',
    'schema'    => 'my_schema',
);
```

The output is:
```
\-These rules must pass for Settings
  |-Key user must be present
  \-Key password must be present
```

Now it mentions not only the missing `user` but also the missing `password` key.

**Comment by wesleyvicthor** [comment]
so guys, [this stupid](https://github.com/wesleyvicthor/Validation/commit/95d3988fc16625c937a17282b57030874dc8f390) code change fix this error but it breaks the template definition for grouped exceptions.
[this test](https://github.com/wesleyvicthor/Validation/blob/master/tests/ValidatorTest.php#L32-L40) breaks in a ridiculous manner.

any thoughts  @Respect/owners ?

**Comment by henriquemoody** [comment]
This fix will be released on Saturday, Oct 24th, with version 1.0:

This test was created to ensure this is working: https://github.com/Respect/Validation/blob/master/tests/integration/issue-179.phpt
